### ISSUE TITLE
number of epoches

### ISSUE THREAD

**Issue by ghost** [SeungjunNah/DeepDeblur_release#15]
Hi!

i wanted to ask what was the number of epochs you used for train?
in the code it is 1 but it doesn't look right

thanks!
Sivan

**Comment by SeungjunNah** [comment]
You may have been confused with two similarly-named variables: opt.nEpochs and opt.epochNumber
opt.epochNumber (default 1) is the (re)starting epoch number when continuing training from saved status.
opt.nEpochs is the total number of epochs to train the model.
The default number of epochs is set to infinity. (see opts.lua: line 42 and 111~113)
To get reasonable results, it takes 4.5 x 10^5 iterations given minibatch size of 4.
That is done when opt.minibatchSize is 4, opt.epochbatchSize is 4000, opt.nEpochs is 450.

**Comment by ghost** [comment]
Hi thanks for the answer!

I am a little confused..
You wrote you have 2103 training images
So i understood you divide the training set to batches of 4 and you do
2103//4 iterations per epoch,
And then you do 450 epochs.
But it sums up to ~500*450 iterations and not 4000*450

Can you please explain more the training part?

Thanks
sivan


בתאריך 18 בדצמ׳ 2017 02:44 AM,‏ "Seungjun Nah" <notifications@github.com>
כתב:

**Comment by SeungjunNah** [comment]
I named the unit of several iterations roughly as 'epoch', but the name is not quite precise in a strict manner.
To make an epoch of training data, all the possible patches from the dataset should be sampled, but simply the number is immensely large. Imagine the number of possible patches of size 256x256 from 720x1080 images. There are about (720-256+1)x(1080-256+1)x 2103 = 8 x 10^8 patches and it takes too long to wait for a single epoch.
This is what usually happens in low-level map regression problems. So, I just sampled a fixed number of random patches from the whole dataset and called the iterations using the set of patches as 'epoch' in the code.

**Comment by ghost** [comment]
Hi thanks again for the answer

In your definition it is 4000*450 iterations, which is not equal to
4.5*10^5.. what im missing?

If i define epoch more traditionally as extracting patch of 256*256 from
all dataset (with random augmentation) so i need to do (4000*450)/(2103//4
) epochs right?

Thanks

בתאריך 18 בדצמ׳ 2017 08:04 AM,‏ "Seungjun Nah" <notifications@github.com>
כתב:

**Comment by SeungjunNah** [comment]
You should divide the epochbatchSize(4000) with minibatchSize(4) to get the number of iterations(1000) in an epoch. That leads to 450x1000 = 4.5x10^5
By the way, note that including augmentation, we use geometric transform(rotation, flip), additive noise (random strength of Gaussian noise), color space (RGB channel shuffle, random saturation change). So the effective number of possible training patches is infinite.

To be more traditional as you say, without augmentation, 450x4000=1.8x10^6 patches are less than all patches for single epoch (8x10^8). So you can say the model doesn't even need a single epoch to be trained. About (1.8x10^6)/(8x10^8)=2.25x10^-3 epochs required.

**Comment by ghost** [comment]
hi, i saw the geometric augmentation and used it.
I am not sure that same image with different augmentation can be considered
as new image because it makes the data dependent.In different epochs it
makes more sense for me.
Thanks ,you've helped me a lot!

2017-12-18 8:37 GMT+02:00 Seungjun Nah <notifications@github.com>:
